Sources/SwiftParser/Patterns.swift

```diff
@@ -354,6 +354,7 @@ extension Parser.Lookahead {
         && !self.at(.keyword(.__owned))
         && !self.at(.keyword(.borrowing))
         && !self.at(.keyword(.consuming))
+        && (!experimentalFeatures.contains(.nonEscapableTypes) || !self.at(.keyword(._resultDependsOn)))
```

Comment:
FWIW I think this would be more readable as follows because that matches the pattern of having a list of and-not expressions.

```suggestion
        && !(experimentalFeatures.contains(.nonEscapableTypes) && self.at(.keyword(._resultDependsOn)))
```